hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/NetworkGetAccountDetailsHandler.java

```diff
@@ -16,54 +16,268 @@
     }
 
     @Override
+    @NonNull
     public QueryHeader extractHeader(@NonNull final Query query) {
         requireNonNull(query);
-        return query.accountDetailsOrThrow().header();
+        return query.accountDetailsOrThrow().headerOrThrow();
     }
 
     @Override
+    @NonNull
     public Response createEmptyResponse(@NonNull final ResponseHeader header) {
         requireNonNull(header);
         final var response = GetAccountDetailsResponse.newBuilder().header(header);
         return Response.newBuilder().accountDetails(response).build();
     }
 
+    @Override
+    public boolean requiresNodePayment(@NonNull ResponseType responseType) {
+        return responseType == ANSWER_ONLY || responseType == ANSWER_STATE_PROOF;
+    }
+
+    @Override
+    public boolean needsAnswerOnlyCost(@NonNull ResponseType responseType) {
+        return COST_ANSWER == responseType;
+    }
+
     @Override
     public void validate(@NonNull final QueryContext context) throws PreCheckException {
         requireNonNull(context);
-        throw new UnsupportedOperationException("Not implemented");
+        final var query = context.query();
+        final var accountStore = context.createStore(ReadableAccountStore.class);
+        final GetAccountDetailsQuery op = query.accountDetailsOrThrow();
+        if (op.hasAccountId()) {
+            final var accountMetadata = accountStore.getAccountById(op.accountIdOrElse(AccountID.DEFAULT));
+            mustExist(accountMetadata, INVALID_ACCOUNT_ID);
+        }
     }
 
     @Override
+    @NonNull
     public Response findResponse(@NonNull final QueryContext context, @NonNull final ResponseHeader header) {
         requireNonNull(context);
         requireNonNull(header);
-        throw new UnsupportedOperationException("Not implemented");
+        final var query = context.query();
+        final var accountStore = context.createStore(ReadableAccountStore.class);
+        final var op = query.accountDetailsOrThrow();
+        final var responseBuilder = GetAccountDetailsResponse.newBuilder();
+        final var account = op.accountIdOrElse(AccountID.DEFAULT);
+        final var ledgerConfig = context.configuration().getConfigData(LedgerConfig.class);
+
+        final var responseType = op.headerOrElse(QueryHeader.DEFAULT).responseType();
+        responseBuilder.header(header);
+        if (header.nodeTransactionPrecheckCode() == OK && responseType != COST_ANSWER) {
+            final var tokensConfig = context.configuration().getConfigData(TokensConfig.class);
+            final var readableTokenStore = context.createStore(ReadableTokenStore.class);
+            final var tokenRelationStore = context.createStore(ReadableTokenRelationStore.class);
+            final var optionalInfo = infoForAccount(
+                    account, accountStore, tokensConfig, readableTokenStore, tokenRelationStore, ledgerConfig);
+            optionalInfo.ifPresent(responseBuilder::accountDetails);
```

Comment:
if(optionalInfo.isEmpty()) we should set response with `FAIL_INVALID` currently.

Reply:
done fixed 
`
     if(optionalInfo.isEmpty()) {
                header.copyBuilder().nodeTransactionPrecheckCode(FAIL_INVALID).build();
                responseBuilder.header(header);
            }
            else {
                optionalInfo.ifPresent(responseBuilder::accountDetails);
            }
`